ports/openssl/unix/CMakeLists.txt

```
@@ -60,7 +60,8 @@ message("CMAKE_C_FLAGS_DEBUG=${CMAKE_C_FLAGS_DEBUG}")
 message("CMAKE_INCLUDE_SYSTEM_FLAG_C=${CMAKE_INCLUDE_SYSTEM_FLAG_C}")
 message("CMAKE_C_OSX_DEPLOYMENT_TARGET_FLAG=${CMAKE_C_OSX_DEPLOYMENT_TARGET_FLAG}")
 
-set(CFLAGS "${CMAKE_C_FLAGS}")
+string(TOUPPER "${CMAKE_BUILD_TYPE}" BUILD_TYPE)
+set(CFLAGS "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_${BUILD_TYPE}}")
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     set(CFLAGS "-Wno-error=unused-command-line-argument ${CMAKE_C_FLAGS}")
```

```suggestion
set(CFLAGS "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_${BUILD_TYPE}}")
if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
    set(CFLAGS "${CFLAGS} -Wno-error=unused-command-line-argument")
```
Otherwise the `CFLAGS` value above will be clobbered if the compiler is Clang (like when building for Android, for example).

---

FTR `CMAKE_CXX_COMPILER_ID` was never initialized because of `project(openssl C)`...

---

Oh, that's a very good catch - yes, we should definitely be using `CMAKE_C_COMPILER_ID`.